src/applications/differential/query/DifferentialRevisionRequiredReviewerActionResultBucket.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+final class DifferentialRevisionRequiredReviewerActionResultBucket
+  extends DifferentialRevisionResultBucket {
+
+  const BUCKETKEY = 'review-action';
+
+  const KEY_REJECTED = 'rejected-review';
+  const KEY_BLOCKING = 'blocking-review';
+  const KEY_SHOULDREVIEW = 'should-review';
+
+  private $objects;
+
+  public function getResultBucketName() {
+    return pht('Bucket by Reviewer Required Action');
+  }
+
+  protected function buildResultGroups(
+    PhabricatorSavedQuery $query,
+    array $objects) {
+
+    $this->objects = $objects;
+
+    $phids = $query->getEvaluatedParameter('reviewerPHIDs');
+    if (!$phids) {
+      throw new Exception(
+        pht(
+          'You can not bucket results by required reviewer action without '.
+          'specifying "Reviewer".'));
+    }
+    $phids = array_fuse($phids);
+
+    // We are only interested in revisions for review
+    $this->filterRevisionsAuthored($phids);
+
+    // Before continuing, throw away any revisions which have been
+    // explicitly resigned from.
+
+    // The goal is to allow users to resign from revisions they don't want to
+    // review to get these revisions off their dashboard, even if there are
+    // other project or package reviewers which they have authority over.
+    $this->filterResigned($phids);
+
+
+    // We also throw away revisions not ready for review.
+    // This includes drafts, accepted, waiting on other reviewers or the author
+    $this->filterDrafts($phids);
+    $this->filterRevisionsAccepted($phids);
+    $this->filterWaitingOnOtherReviewers($phids);
+    $this->filterWaitingOnAuthors($phids);
+
+    $groups = array();
+
+    // it would be better if this filter inculded only those upated since rejected
```

Comment:
a. typos: `included`, `updated`
b. what does this mean? "it would be better if the filter only returned revisions which have been updated since they were rejected by a reviewer, but ..." ?